File: python/verifySomeBehavior/birthdayservice.py
class BirthdayService(object):

    def __init__(self, client_repository, mailer):
        self.mailer = mailer
        self.client_repository = client_repository

    def greeting(self, name):
        if self.client_repository.birthdayIsTodayFor(name):
            greeting = 'Happy birthday ' + name + '!'
        else:
            greeting = ''

        self.mailer.send(greeting)

File: php/src/spy/BirthdayService.php
<?php


namespace mock_and_stub_introduction\spy;

class BirthdayService {
    
    private $clientRepository;
    private $mailer;

    public function __construct($clientRepository, Mailer $mailer) {
        $this->clientRepository = $clientRepository;
        $this->mailer = $mailer;
    }
    
    public function greetingFor($name) {
        $greeting = "";
        if ($this->clientRepository->birthDayIsTodayFor($name)) {
            $greeting = "Happy birthday $name!";
        }
        else {
            $greeting = "Good morning $name.";
        }

        $this->mailer->send($greeting);
    }
    
}

?>
